services/github/github-go-mod.service.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { renderVersionBadge } = require('../version')
+const { ConditionalGithubAuthV3Service } = require('./github-auth-service')
+const { fetchRepoContent } = require('./github-common-fetch')
+const { documentation } = require('./github-helpers')
+const { InvalidResponse } = require('..')
+
+const queryParamSchema = Joi.object({
+  filename: Joi.string(),
+}).required()
+
+const goVersionRegExp = new RegExp('^go (.+)$', 'm')
+
+module.exports = class GithubGoModGoVersion extends ConditionalGithubAuthV3Service {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'github/go-mod/go-version',
+      pattern: ':user/:repo/:branch*',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub go.mod Go version',
+        pattern: ':user/:repo',
+        namedParams: { user: 'gohugoio', repo: 'hugo' },
+        staticPreview: this.render({ version: '1.12' }),
+        documentation,
+      },
+      {
+        title: 'GitHub go.mod Go version (branch)',
+        pattern: ':user/:repo/:branch',
+        namedParams: {
+          user: 'gohugoio',
+          repo: 'hugo',
+          branch: 'master',
+        },
+        staticPreview: this.render({ version: '1.12', branch: 'master' }),
+        documentation,
+      },
+      {
+        title: 'GitHub go.mod Go version (subfolder of monorepo)',
+        pattern: ':user/:repo',
+        namedParams: { user: 'golang', repo: 'go' },
+        queryParams: { filename: 'src/go.mod' },
+        staticPreview: this.render({ version: '1.14' }),
+        documentation,
+      },
```

Comment:
Can we update the example listings to include supporting a custom file path/subfolder with a specific branch too?

Reply:
Done.